compiler/qsc_rca/src/errors.rs

```diff
@@ -51,6 +51,20 @@ pub enum Error {
     #[diagnostic(code("Qsc.CapabilitiesCk.UseOfDynamicQubit"))]
     UseOfDynamicQubit(#[label] Span),
 
+    #[error("cannot use a dynamic result")]
+    #[diagnostic(help(
+        "using a mutable Result variable whose value depends on a measurement result is not supported by the current target"
```

Comment:
that's a good point... at first this error was limited to mutable Q# variables, but it turns out it also needs to apply to code with immutable binding, like:
```qsharp
let result = if M(q) == Zero { One } else { M(q) };
```
I'll update the text to not use that term, since it's no longer specific to that.